docs/appendix-A.rst

```diff
@@ -0,0 +1,1783 @@
+.. Copyright (c) 2017-2019, Lawrence Livermore National Security, LLC and
+   other Shroud Project Developers.
+   See the top-level COPYRIGHT file for details.
+
+   SPDX-License-Identifier: (BSD-3-Clause)
+
+.. All of the examples are ordered as
+   C or C++ code from users library
+   YAML input
+   C++ wrapper
+   Fortran interface
+   Fortran wrapper
+   Fortran example usage
+   C++ example usage
+
+Sample Fortran Wrappers
+=======================
+
+This chapter gives details of the generated code.
+It's intended for users who want to understand the details
+of how the wrappers are created.
+
+All of these examples are derived from tests in the ``regression``
```

Comment:
These examples are great!

It might be helpful if these examples would distinguish between the user code, and the shroud generated code.

Reply:
I added more comments to make it easer to tell the difference.  The general pattern is user's library at the top, usage example at the bottom, and the rest is generated.